triples-generator/src/test/scala/io/renku/triplesgenerator/events/categories/triplesgenerated/projectinfo/ProjectEventsFinderSpec.scala

```diff
@@ -0,0 +1,191 @@
+          endpointName,
+          *,
+          maybeAccessToken
+        )
+        .returning(expectation.pure[IO])
+
+      finder.find(project, page).value.unsafeRunSync() shouldBe (events.map(_.asNormalPushEvent),
+                                                                 PagingInfo(None, None)
+      ).asRight
+    }
+
+    // mapResponse Tests
+
+    "take from commitTo and skip commitFrom if both commitIds exist on the event" in new TestCase {
+      val commitFrom = commitIds.generateSome
+      val commitTo   = commitIds.generateSome
+      val event = pushEvents.generateOne
+        .forMember(member)
+        .forProject(project)
+        .copy(maybeCommitFrom = commitFrom, maybeCommitTo = commitTo)
+      val events = Random.shuffle(event :: pushEvents.generateNonEmptyList().toList)
+
+      mapResponse(Status.Ok, Request[IO](), Response[IO]().withEntity(events.asJson.noSpaces))
+        .unsafeRunSync() shouldBe (events.map(
+        _.asNormalPushEvent
+      ),
+      PagingInfo(None, None))
+    }
+
+    "take commitFrom if commitTo doesn't exist on the event" in new TestCase {
+      val commitFrom = commitIds.generateSome
+      val event = pushEvents.generateOne
+        .forMember(member)
+        .forProject(project)
+        .copy(maybeCommitFrom = commitFrom, maybeCommitTo = None)
+      val events = Random.shuffle(event :: pushEvents.generateNonEmptyList().toList)
+
+      mapResponse(Status.Ok, Request[IO](), Response[IO]().withEntity(events.asJson.noSpaces))
+        .unsafeRunSync() shouldBe (events.map(
+        _.asNormalPushEvent
+      ),
+      PagingInfo(None, None))
+    }
+
+    "parse the response headers for 'next page' and 'total pages'" in new TestCase {
+      val events     = pushEvents.generateNonEmptyList().toList
+      val nextPage   = nonNegativeInts().generateOne.value
+      val totalPages = nonNegativeInts().generateOne.value
+      val headers = Headers(
+        List(Header.Raw(ci"X-Next-Page", nextPage.toString()), Header.Raw(ci"X-Total-Pages", totalPages.toString()))
+      )
+
+      mapResponse(Status.Ok, Request[IO](), Response[IO]().withEntity(events.asJson.noSpaces).withHeaders(headers))
+        .unsafeRunSync() shouldBe (events.map(_.asNormalPushEvent), PagingInfo(nextPage.some, totalPages.some))
+    }
+
+    "return an empty List if project NotFound" in new TestCase {
+      mapResponse(Status.NotFound, Request[IO](), Response[IO]()).unsafeRunSync() shouldBe
+        (List.empty[PushEvent], PagingInfo(None, None))
+    }
+
+    "fail the interpretation if response is Unauthorized" in new TestCase {
+
+      intercept[Exception] {
+        mapResponse(Status.Unauthorized, Request[IO](), Response[IO]()).unsafeRunSync() shouldBe
+          (None, PagingInfo(None, None))
+      }
+    }
+  }
+
+  private lazy val requestTimeout = 2 seconds
+
+  private trait TestCase {
+    
+    val project = Project(projectIds.generateOne, projectPaths.generateOne)
```

Comment:
I guess we could use `consumerProjects` from the `ConsumersModelGenerators` if we like

Reply:
In the MemberEmailFinder, we have a class `private final case class Project(id: projects.Id, path: projects.Path)` Is there a reason to have this separate class?